### DISCUSSION TITLE
Combine custom component with text

### DISCUSSION THREAD

**Discussion by damnslow** [GrapesJS/grapesjs#3821] [Q&A]
I know this is possible, but I'm unable to find docs around it.

How can I have a custom component with some editable field? For example: I have added a new block with styled nested divs and spans. But can I have editable fields by default instead of dragging a text component inside it?

**Comment by artf** [comment]
If you're using HTML string you can pass properties to components via `data-gjs-*` attributes
```js
// your block
content: `<div>
  <div data-gjs-type="text">...</div>
</div>`
```
